plugins/arrow-flight-rpc/src/main/java/org/apache/arrow/flight/OSFlightClient.java

```diff
@@ -0,0 +1,950 @@
+/*
```

Comment:
Can you point to the exact SSL changes in this class and in `OSFlightServer`? This is a big class and hard to find the differences between this and the original `FlightClient`

Reply:
@rishabhmaurya so you only need to clone `Builder` class (sadly) and not a whole `FlightClient`, I assume the same applies to the `FlightServer`.  We could move all of them under `org.opensearch.arrow.flight` package as well.

Reply:
@reta  builder class is a final class, so I had to override everything. 
I have tried to simplify the logic of cloning in the latest version, the only change FlightServer and FlightClient have is additional method to hook `configureBuilder()` and making Builder classes non-final. Let me know what you think

Reply:
> @reta builder class is a final class, so I had to override everything.

That is fine, missed the fact that FlightClient (and FlightServer) constructors are package private ((( Could you please submit pull request to Apache Arrow? Let's see how it is going to be accepted, I would prefer to avoid copying such a massive amount of code.

Reply:
yes, I will raise the PR with them to separate the building logic from client and make it extensible. We ultimately will combine grpc and flight server and then we might need more such customization in builder class.
@reta Please let me know if we should let this PR in and I will try to work getting arrow PR merged separately.

Reply:
@rishabhmaurya pushed a change for `OSFlightClient` (this is what I meant in https://github.com/opensearch-project/OpenSearch/pull/16962/files#r1951874973), curious what do you think about it (shoveled off a large chunk of code). If it does make sense, could you please do the same for `OSFlightServer` (just keep the builder)? Thank you (and my apologies for the delays).

Reply:
@reta It makes sense for FlightClient, thanks for addressing it. For FlightServer, the constructor has private access, thus its not possible to just build it by cloning the builder. Let me know what do you think?

Reply:
@rishabhmaurya we need to be a little bit more creative there, pushed the change, please let me know what do you think, thank you.

Reply:
@reta that's indeed very creative, shove off a ton of redundant code. Thank you.